osquery/tables/system/windows/pipes.cpp

```diff
@@ -71,6 +111,43 @@ QueryData genPipes(QueryContext& context) {
                            : "PIPE_TYPE_BYTE";
     r["flags"] = end + ',' + type;
 
+    // Get a pointer to the existing DACL.
+    PACL dacl = nullptr;
+    auto result = GetNamedSecurityInfo(pipeHandle,
+                                       SE_FILE_OBJECT,
+                                       DACL_SECURITY_INFORMATION,
+                                       nullptr,
+                                       nullptr,
+                                       &dacl,
+                                       nullptr,
+                                       nullptr);
+    if (ERROR_SUCCESS != result) {
+      VLOG(1) << "GetExplicitEnteriesFromAcl Error " << result;
+      continue;
+    }
+
+    // get list of ACEs from DACL pointer
+    unsigned long aceCount = 0;
+    PEXPLICIT_ACCESS aceList = nullptr;
+    result = GetExplicitEntriesFromAcl(dacl, &aceCount, &aceList);
+    if (ERROR_SUCCESS != result) {
+      VLOG(1) << "GetExplicitEnteriesFromAcl Error " << result;
```

Comment:
Do these returns need to close the pipeHandle since they aren't reaching the end of the function?

Reply:
same than https://github.com/facebook/osquery/blob/8f7ea728d269b6a1096b39cb5b05573b72e7190e/osquery/tables/system/windows/ntfs_acl_permissions.cpp